code/modules/mob/living/simple_animal/hostile/terror_spiders/terror_ai.dm

```diff
@@ -92,10 +90,18 @@
 // --------------------- TERROR SPIDERS: AI BEHAVIOR CODE -------------------------
 // --------------------------------------------------------------------------------
 
+
 /mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_action()
-	if (stat || ckey)
-		return ..()
-	if(AIStatus != AI_OFF && !target)
+	if(target)
+		CreatePath(target)
+	return ..()
+
+/mob/living/simple_animal/hostile/poison/terror_spider/handle_automated_movement()
+	// Putting the main terror spider AI code in handle_automated_movement() rather than handle_automated_action() ensures it will still run when the spider AIStatus == AI_IDLE
+	// This is necessary for the terror spiders in the away mission to work properly.
+	if(stat || ckey || AIStatus != AI_IDLE)
```

Comment:
there's no way that `handle_automated_movement` gets called on something with that's dead or has a ckey.

Reply:
Updated to remove those two checks.